packages/flutter/lib/src/rendering/text_boundary.dart

```diff
@@ -0,0 +1,174 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:characters/characters.dart' show CharacterRange;
+import 'package:flutter/services.dart';
```

Comment:
what's the reason this is put in rendering (as opposed to, for example, services)?

Reply:
services probably makes more sense, will do